framework/execution_framework/plugins/org.eclipse.gemoc.executionframework.extensions.sirius/src/org/eclipse/gemoc/executionframework/extensions/sirius/modelloader/DefaultModelLoader.java

```diff
@@ -304,76 +304,77 @@ protected void doExecute() {
-					protected void doExecute() {
-						boolean hasADebugLayer = false;
-						Set<Layer> layers = new HashSet<Layer>();
-						layers.addAll(diagram.getDescription().getAdditionalLayers());
-						Collection<Viewpoint> selectedVp = session.getSelectedViewpoints(true);
-						for (Viewpoint vp : selectedVp) {
-							for (RepresentationExtensionDescription extension : vp.getOwnedRepresentationExtensions()) {
-								if (extension instanceof DiagramExtensionDescription) {
-									layers.addAll(((DiagramExtensionDescription) extension).getLayers());
+						protected void doExecute() {
+							boolean hasADebugLayer = false;
+							Set<Layer> layers = new HashSet<Layer>();
+							layers.addAll(diagram.getDescription().getAdditionalLayers());
+							Collection<Viewpoint> selectedVp = session.getSelectedViewpoints(true);
+							for (Viewpoint vp : selectedVp) {
+								for (RepresentationExtensionDescription extension : vp.getOwnedRepresentationExtensions()) {
+									if (extension instanceof DiagramExtensionDescription) {
+										layers.addAll(((DiagramExtensionDescription) extension).getLayers());
+									}
 								}
 							}
-						}
-						for (Layer l : layers) {
-							String descName = diagram.getDescription().getName();
-							String layerName = l.getName();
-							boolean mustBeActiveForDebug = AbstractDSLDebuggerServices.LISTENER
-									.isRepresentationToRefresh(context.getRunConfiguration().getDebugModelID(),
-											descName, layerName)
-									|| layerName.equalsIgnoreCase("Debug");
-							boolean mustBeActiveForAnimation = AbstractGemocAnimatorServices.ANIMATOR
-									.isRepresentationToRefresh(descName, layerName)
-									|| layerName.equalsIgnoreCase("Animation");
-							boolean mustBeActive = mustBeActiveForAnimation || mustBeActiveForDebug;
-							hasADebugLayer = hasADebugLayer || mustBeActiveForDebug;
-							if (mustBeActive && !diagram.getActivatedLayers().contains(l)) {
-								ChangeLayerActivationCommand c = new ChangeLayerActivationCommand(editingDomain,
-										diagram, l, openEditorSubMonitor.newChild(1));
-								c.execute();
+							for (Layer l : layers) {
+								String descName = diagram.getDescription().getName();
+								String layerName = l.getName();
+								boolean mustBeActiveForDebug = AbstractDSLDebuggerServices.LISTENER
+										.isRepresentationToRefresh(context.getRunConfiguration().getDebugModelID(),
+												descName, layerName)
+										|| layerName.equalsIgnoreCase("Debug");
+								boolean mustBeActiveForAnimation = AbstractGemocAnimatorServices.ANIMATOR
+										.isRepresentationToRefresh(descName, layerName)
+										|| layerName.equalsIgnoreCase("Animation");
+								boolean mustBeActive = mustBeActiveForAnimation || mustBeActiveForDebug;
+								hasADebugLayer = hasADebugLayer || mustBeActiveForDebug;
+								if (mustBeActive && !diagram.getActivatedLayers().contains(l) && !diagram.getActivatedTransientLayers().contains(l)) {
```

Comment:
this is the second line that has changed (most other changes are due to the new indentation)